org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/ast/LambdaExpression.java

```diff
@@ -560,9 +560,23 @@ private void analyzeExceptions() {
 		boolean oldAnalyseResources = compilerOptions.analyseResourceLeaks;
 		compilerOptions.analyseResourceLeaks = false;
 		try {
+			int firstLocalLocal = this.scope.outerMostMethodScope().analysisIndex;
```

Comment:
Strictly speaking `analysisIndex` points to the next _free_ slot in locals. But a variable name `firstLambdaLocalIfThereWereAny` doesn't look better, nor is `indexAfterTrulyOuterLocals` in for a beauty award :)